### DISCUSSION TITLE
Asynchronous multiple data sends in single message

### DISCUSSION THREAD

**Discussion by karthi8185** [open-telemetry/opentelemetry-go#4254] [Q&A] [answered]
Hi, 

 As per the below example. Can i say the two data (heapAlloc, gcCount) transmitted in a single message to Otel Collector?
 Kindly confirm how the data transfer processing happening in otel API. 

package main

import (
	"context"
	"fmt"
	"runtime"

	"go.opentelemetry.io/otel"
	"go.opentelemetry.io/otel/metric"
)

func main() {
	meter := otel.Meter("go.opentelemetry.io/otel/metric#MultiAsyncExample")

	// This is just a sample of memory stats to record from the Memstats
	heapAlloc, _ := meter.Int64ObservableUpDownCounter("heapAllocs")
	gcCount, _ := meter.Int64ObservableCounter("gcCount")

	_, err := meter.RegisterCallback(
		func(_ context.Context, o metric.Observer) error {
			memStats := &runtime.MemStats{}
			// This call does work
			runtime.ReadMemStats(memStats)

			o.ObserveInt64(heapAlloc, int64(memStats.HeapAlloc))
			o.ObserveInt64(gcCount, int64(memStats.NumGC))

			return nil
		},
		heapAlloc,
		gcCount,
	)
	if err != nil {
		fmt.Println("Failed to register callback")
		panic(err)
	}
}

**Comment by pellared** [comment, marked as answer]
> Can i say the two data (heapAlloc, gcCount) transmitted in a single message to Otel Collector?

In general (unless you have not done anything strange) yes. But you have not shared the code how you setup the OTLP exporter and meter provider.

**Reply by karthi8185** [reply]
yes, i created the OTLP exporter and meter provider.

**Reply by karthi8185** [reply]
Thanks @pellared,  So it helps to send multiple data in a single message at a time
